pkg/logs/client/http/destination.go

```diff
@@ -218,11 +251,17 @@ func CheckConnectivity(endpoint config.Endpoint) config.HTTPConnectivity {
 	// Lower the timeout to 5s because HTTP connectivity test is done synchronously during the agent bootstrap sequence
 	destination := newDestination(endpoint, JSONContentType, ctx, time.Second*5, 0)
 	log.Infof("Sending HTTP connectivity request to %s...", destination.url)
-	err := destination.Send(emptyPayload)
+	err := destination.send(emptyPayload)
 	if err != nil {
 		log.Warnf("HTTP connectivity failure: %v", err)
 	} else {
 		log.Info("HTTP connectivity successful")
 	}
 	return err == nil
 }
+
+func (d *Destination) waitForBackoff() {
+	ctx, cancel := context.WithDeadline(d.destinationsContext.Context(), d.blockedUntil)
```

Comment:
Is there a reason `context.WithDeadline` is prefered over `timer.NewTimer` in this case? (mostly just curious)

Reply:
`WithDeadline` avoids backoff delay when the parent context is cancelled. This happens during agent shutdown while backoff is active, see https://github.com/DataDog/datadog-agent/blob/0f2b556fc40a3a780c9ac932a93b397fd176ac49/pkg/logs/agent.go#L163-L167

Reply:
nice makes sense. thanks for clarifying :)